graylog2-server/src/test/java/org/graylog2/plugin/inputs/transports/AbstractTcpTransportTest.java

```diff
@@ -122,15 +119,15 @@ public void getChildChannelHandlersFailsIfTempDirDoesNotExist() throws IOExcepti
         final AbstractTcpTransport transport = new AbstractTcpTransport(
             configuration, throughputCounter, localRegistry, eventLoopGroup, eventLoopGroupFactory, nettyTransportConfiguration, tlsConfiguration) {};
 
-        expectedException.expect(IllegalStateException.class);
-        expectedException.expectMessage("Couldn't write to temporary directory: " + tmpDir.getAbsolutePath());
+        Throwable exception = assertThrows(IllegalStateException.class, () ->
 
-        transport.getChildChannelHandlers(input);
+            transport.getChildChannelHandlers(input));
+        org.hamcrest.MatcherAssert.assertThat(exception.getMessage(), containsString("Couldn't write to temporary directory: " + tmpDir.getAbsolutePath()));
```

Comment:
While correct, the more elegant conversion would be something like this:
```
        assertThatThrownBy(() -> transport.getChildChannelHandlers(input))
                .isInstanceOf(IllegalStateException.class)
                .hasMessageContaining("Couldn't write to temporary directory: " + tmpDir.getAbsolutePath());
```
But that would be the gold-plated solution.